### ISSUE TITLE
terminate called after throwing an instance of 'luabridge::LuaException' 

### ISSUE THREAD

**Issue by sodomon2** [TES3MP/openmw-tes3mp#588]
Hi,I have had a problem playing tes3mp which causes the server to crash and the client to lose all connection.

this is the error

```
[2021-07-13 06:04:28] [INFO]: [Script]: The world time hour is now 10
[2021-07-13 06:04:35] [INFO]: Received ID_OBJECT_ACTIVATE from Sieger
[2021-07-13 06:04:35] [INFO]: [Script]: Called "OnObjectActivate" for Sieger (1) and cell Balmora (-3, -2)
[2021-07-13 06:04:35] [ERR]: ./CoreScripts/scripts/eventHandler.lua:1223: attempt to concatenate a nil value
[2021-07-13 06:04:35] [ERR]: [Script]: Server crash from script error!
terminate called after throwing an instance of 'luabridge::LuaException'
  what():  ./CoreScripts/scripts/eventHandler.lua:1223: attempt to concatenate a nil value
Aborted


```

**Comment by davidcernat** [comment]
Can you provide your server/data/cell/-3, -2.json file?

**Comment by sodomon2** [comment]
@davidcernat Do you mean Balmora (-3, -2).json? 
if so, here is the content and the file

```
{
  "lastVisit":[],
  "recordLinks":[],
  "packets":{
    "spawn":[],
    "scale":[],
    "trap":[],
    "cellChangeFrom":[],
    "equipment":[],
    "delete":[],
    "lock":[],
    "position":[],
    "miscellaneous":[],
    "cellChangeTo":[],
    "statsDynamic":[],
    "place":[],
    "clientScriptLocal":[],
    "doorState":[],
    "death":[],
    "state":[],
    "container":[],
    "ai":[],
    "actorList":[]
  },
  "entry":{
    "creationTime":1626309186,
    "description":"Balmora (-3, -2)"
  },
  "objectData":[]
}
```

[Balmora (-3, -2).txt](https://github.com/TES3MP/openmw-tes3mp/files/6820074/Balmora.-3.-2.txt)

**Comment by Sarevalak** [comment]
Server is using `Cell::getDescription()` method to get external cell coordinates, but method was changed
https://github.com/TES3MP/openmw-tes3mp/commit/aa5161f99e1f4f87d9934733b9102d921500f850#diff-e6754e64fe0ddb74e45c31c8598ae55282f5b6397198414c72b04326a68a6e27

**Comment by sodomon2** [comment]
> If the cell filename looks like this, then I assume that server is using `Cell::getDescription()` method to get external cell coordinates, but method was changed
> [aa5161f#diff-e6754e64fe0ddb74e45c31c8598ae55282f5b6397198414c72b04326a68a6e27](https://github.com/TES3MP/openmw-tes3mp/commit/aa5161f99e1f4f87d9934733b9102d921500f850#diff-e6754e64fe0ddb74e45c31c8598ae55282f5b6397198414c72b04326a68a6e27)

And do you have any way of knowing how to fix it?

**Comment by Sarevalak** [comment]
It is possible reason of crash. I don't know what version of the code your application build is based on to check if this is so.
If the assumption about reason of crash is right, then the easiest way to fix this issue is to revert the changes that i mentioned earlier. But this is not the better way to do it.

**Comment by sodomon2** [comment]
> It is possible reason of crash. I don't know what version of the code your application build is based on to check if this is so.
> If the assumption about reason of crash is right, then the easiest way to fix this issue is to revert the changes that i mentioned earlier. But this is not the better way to do it.

I am using version 0.7.1
Ok,I will try tand revert the changes and report back if it gives any error.

**Comment by Sarevalak** [comment]
You are using your own build?

**Comment by sodomon2** [comment]
> You are using your own build?

Yes, I built with these instructions and I had to apply this patch 

[tes3mp.patch](https://github.com/TES3MP/openmw-tes3mp/files/6831495/tes3mp.txt)
```
cmake -G Ninja -DCMAKE_BUILD_TYPE=MinSizeRel ../ -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DRakNet_LIBRARY_DEBUG=/usr/lib/libRakNetLibStatic.a -DCMAKE_CXX_FLAGS="-fpermissive"
```

**Comment by sodomon2** [comment]
> You are using your own build?

Yes, I built with these instructions and I had to apply this patch 

[tes3mp.patch](https://github.com/TES3MP/openmw-tes3mp/files/6831495/tes3mp.txt)
```
cmake -G Ninja -DCMAKE_BUILD_TYPE=MinSizeRel ../ -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DRakNet_LIBRARY_DEBUG=/usr/lib/libRakNetLibStatic.a -DCMAKE_CXX_FLAGS="-fpermissive"
```

**Comment by sodomon2** [comment]
> You are using your own build?

Yes, I built with these instructions and I had to apply this patch 

[tes3mp.patch](https://github.com/TES3MP/openmw-tes3mp/files/6831495/tes3mp.txt)
```
cmake -G Ninja -DCMAKE_BUILD_TYPE=MinSizeRel ../ -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DRakNet_LIBRARY_DEBUG=/usr/lib/libRakNetLibStatic.a -DCMAKE_CXX_FLAGS="-fpermissive"
```
